compiler/fir/checkers/checkers-component-generator/src/org/jetbrains/kotlin/fir/checkers/generator/diagnostics/FirDiagnosticsList.kt

```diff
@@ -483,7 +483,7 @@ object DIAGNOSTICS_LIST : DiagnosticList() {
     val FUNCTIONS by object : DiagnosticGroup("Functions") {
         val ABSTRACT_FUNCTION_IN_NON_ABSTRACT_CLASS by error<KtFunction>(PositioningStrategy.MODALITY_MODIFIER) {
             parameter<FirMemberDeclaration>("function")
-            parameter<FirMemberDeclaration>("containingClass") // TODO use FirClass instead of FirMemberDeclaration
+            parameter<FirRegularClass>("containingClass")
```

Comment:
I'd prefer to have `FirClass` here and below. At this moment we don't report this error on anonymous object members, however it's just a bug.